### ISSUE TITLE
Vaadin builds are suddenly severely broken

### ISSUE THREAD

**Issue by mmoser18** [vaadin/flow#11162]
Since three days I have MAJOR issues building and esp. executing Vaadin apps!
First I though I had somehow corrupted my project setup, as I ran into odd issues re. recursion during frontend compilation and what not. Then I reverted my project to an earlier version - same error. I reinstalled my project directory from an earlier backup (where it HAD been working!). That version built OK but hung after startup during the front-end compilation.

By now I regressed to projects that I freshly create and download from the Vaadin home page's application generator (https://start.vaadin.com/). I can build that application and it starts up fine, but when I direct my browser to it I get:

```
Webpack build failed with errors:
ERROR in Entry module not found: Error: Can't resolve '@vaadin/flow-frontend/VaadinDevmodeGizmo.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\frontend'
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/comboBoxConnector-es6.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 6:0-56
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/comboBoxConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 83:0-52
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/contextMenuConnector-es6.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 7:0-59
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/contextMenuConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 84:0-55
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/datepickerConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 8:0-54
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/dndConnector-es6.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 9:0-51
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/dndConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 85:0-47
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/flow-component-renderer.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 10:0-58
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/gridConnector-es6.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 11:0-52
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/gridConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 86:0-48
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/gridProConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 12:0-51
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/ironListConnector-es6.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 13:0-56
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/ironListConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 87:0-52
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/ironListStyles.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 14:0-49
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/menubarConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 15:0-51
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/selectConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 88:0-50
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/timepickerConnector.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 16:0-54
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/vaadin-big-decimal-field.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 17:0-59
ERROR in ../target/frontend/generated-flow-imports.js
Module not found: Error: Can't resolve '@vaadin/flow-frontend/vaadin-grid-flow-selection-column.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
 @ ../target/frontend/generated-flow-imports.js 18:0-68
```

Why do all my Vaadin applications all of a sudden not work anymore?

Meanwhile I generated and downloaded "Hello World"-apps generated for Vaadin versions 14, 18 and 20 and Java v8 and v11. 
I rebooted my system. I removed the .vaadin sub-directory from my home directory (suspecting it could hold corrupted data). I deleted my maven repository to force a re-download of all libraries. Builds work OK but still no successful execution! 

What is going on here? Not a single Vaadin application comes up and works properly any more as it used to! 
What could be possibly wrong here??? Is there any Vaadin setup or config stored somewhere that might be corrupted? I am really desperate by now!

**Comment by knoobie** [comment]
Did you try the "Vaadin dance" https://mvysny.github.io/Vaadin-troubleshooting/ ?

**Comment by mmoser18** [comment]
Hi @knoobie - thanks for responding. Highly appreciated! Those instructions gave me some direction and I followed them. However, I was still not able to get my feet onto the ground. :-(

After I did the "vaadin dance" I issued a `mvn clean install` to the example application that I had download from the Vaadin application generator. The application indeed built fine but then executing it (using `java -jar myapp-1.0-SNAPSHOT.jar`) I get:

```
------------------ Starting Frontend compilation. ------------------
2021-06-05 16:09:59.439  INFO 23852 --- [onPool-worker-3] dev-webpack                              : Running webpack to compile frontend resources. This may tak                                                                                                                e a moment, please stand by...
2021-06-05 16:10:00.430  INFO 23852 --- [           main] o.hibernate.annotations.common.Version   : HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
2021-06-05 16:10:02.139  INFO 23852 --- [           main] org.hibernate.dialect.Dialect            : HHH000400: Using dialect: org.hibernate.dialect.H2Dialect
2021-06-05 16:10:06.727  INFO 23852 --- [           main] o.h.e.t.j.p.i.JtaPlatformInitiator       : HHH000490: Using JtaPlatform implementation: [org.hibernate                                                                                                                .engine.transaction.jta.platform.internal.NoJtaPlatform]
2021-06-05 16:10:06.793  INFO 23852 --- [           main] j.LocalContainerEntityManagerFactoryBean : Initialized JPA EntityManagerFactory for persistence unit '                                                                                                                default'
2021-06-05 16:10:10.110  WARN 23852 --- [           main] JpaBaseConfiguration$JpaWebConfiguration : spring.jpa.open-in-view is enabled by default. Therefore, d                                                                                                                atabase queries may be performed during view rendering. Explicitly configure spring.jpa.open-in-view to disable this warning
2021-06-05 16:10:11.724  INFO 23852 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2021-06-05 16:10:16.041  INFO 23852 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2021-06-05 16:10:16.152  INFO 23852 --- [           main] com.example.application.Application      : Started Application in 174.024 seconds (JVM running for 175                                                                                                                .261)
2021-06-05 16:10:23.799  INFO 23852 --- [nio-8080-exec-3] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2021-06-05 16:10:23.800  INFO 23852 --- [nio-8080-exec-3] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2021-06-05 16:10:23.816  INFO 23852 --- [nio-8080-exec-3] o.s.web.servlet.DispatcherServlet        : Completed initialization in 3 ms
2021-06-05 16:10:24.057  INFO 23852 --- [nio-8080-exec-3] c.v.f.s.DefaultDeploymentConfiguration   :
Vaadin is running in DEBUG MODE.
When deploying application for production, remember to disable debug features. See more from https://vaadin.com/docs/
2021-06-05 16:10:29.056 ERROR 23852 --- [        webpack] dev-webpack                              : ERROR in Entry module not found: Error: Can't resolve '@vaa                                                                                                                din/flow-frontend/VaadinDevmodeGizmo.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\frontend'
2021-06-05 16:10:29.061 ERROR 23852 --- [        webpack] dev-webpack                              : ERROR in ./themes/myapp/styles.css
2021-06-05 16:10:29.071 ERROR 23852 --- [        webpack] dev-webpack                              : Module build failed (from ../node_modules/css-loader/dist/c                                                                                                                js.js):
2021-06-05 16:10:29.076 ERROR 23852 --- [        webpack] dev-webpack                              : Error: Can't resolve 'lumo-css-framework/all-classes.css' i                                                                                                                n 'U:\Documents\eclipse\workspace_Vaadin\my-app\frontend\themes\myapp'
2021-06-05 16:10:29.078 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\Resolver.js:209:21
2021-06-05 16:10:29.079 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\Resolver.js:285:5
2021-06-05 16:10:29.082 ERROR 23852 --- [        webpack] dev-webpack                              :     at eval (eval at create (U:\Documents\eclipse\workspace                                                                                                                _Vaadin\my-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
2021-06-05 16:10:29.098 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\UnsafeCachePlugin.js:44:7
2021-06-05 16:10:29.101 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\Resolver.js:285:5
2021-06-05 16:10:29.104 ERROR 23852 --- [        webpack] dev-webpack                              :     at eval (eval at create (U:\Documents\eclipse\workspace                                                                                                                _Vaadin\my-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
2021-06-05 16:10:29.105 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\Resolver.js:285:5
2021-06-05 16:10:29.109 ERROR 23852 --- [        webpack] dev-webpack                              :     at eval (eval at create (U:\Documents\eclipse\workspace                                                                                                                _Vaadin\my-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:27:1)
2021-06-05 16:10:29.110 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\DescriptionFilePlugin.js:67:43
2021-06-05 16:10:29.113 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\Resolver.js:285:5
2021-06-05 16:10:29.124 ERROR 23852 --- [        webpack] dev-webpack                              :     at eval (eval at create (U:\Documents\eclipse\workspace                                                                                                                _Vaadin\my-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:28:1)
2021-06-05 16:10:29.135 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\ModuleKindPlugin.js:30:40
2021-06-05 16:10:29.137 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\Resolver.js:285:5
2021-06-05 16:10:29.139 ERROR 23852 --- [        webpack] dev-webpack                              :     at eval (eval at create (U:\Documents\eclipse\workspace                                                                                                                _Vaadin\my-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
2021-06-05 16:10:29.142 ERROR 23852 --- [        webpack] dev-webpack                              :     at U:\Documents\eclipse\workspace_Vaadin\my-app\node_mo                                                                                                                dules\enhanced-resolve\lib\Resolver.js:285:5
2021-06-05 16:10:29.144 ERROR 23852 --- [        webpack] dev-webpack                              :     at eval (eval at create (U:\Documents\eclipse\workspace                                                                                                                _Vaadin\my-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:16:1)
2021-06-05 16:10:29.162 ERROR 23852 --- [        webpack] dev-webpack                              :  @ ./themes/myapp/myapp.generated.js 68:0-37 81:20-29
2021-06-05 16:10:29.164 ERROR 23852 --- [        webpack] dev-webpack                              :  @ ../target/frontend/themes/theme-generated.js
2021-06-05 16:10:29.165 ERROR 23852 --- [        webpack] dev-webpack                              :  @ ../target/frontend/generated-flow-imports.js
2021-06-05 16:10:29.167 ERROR 23852 --- [        webpack] dev-webpack                              : ERROR in ../target/frontend/generated-flow-imports.js
2021-06-05 16:10:29.169 ERROR 23852 --- [        webpack] dev-webpack                              : Module not found: Error: Can't resolve '@vaadin/flow-fronte                                                                                                                nd/comboBoxConnector-es6.js' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\target\frontend'
2021-06-05 16:10:29.170 ERROR 23852 --- [        webpack] dev-webpack                              :  @ ../target/frontend/generated-flow-imports.js 6:0-56
2021-06-05 16:10:29.172 ERROR 23852 --- [        webpack] dev-webpack                              : ERROR in ../target/frontend/generated-flow-imports.js
2021-06-05 16:10:29.183 ERROR 23852 --- [        webpack] dev-webpack                              : Module not found: Error: Can't resolve '@vaadin/flow-fronte  
... (etc.)
```
(full log of that run attached: log_running_my_app.txt). 

I then thought that maybe I should give it a try with the -Pproduction flag included, so that it tries to resolve all dependencies at compile time.

So, I did the Vaadin dance again and then issued a `mvn clean install -Pproduction` but that failed already during the build-frontend step:

```
...
ERROR in ./themes/myapp/styles.css
Module build failed (from ../node_modules/css-loader/dist/cjs.js):
Error: Can't resolve 'lumo-css-framework/all-classes.css' in 'U:\Documents\eclipse\workspace_Vaadin\my-app\frontend\themes\myapp'
    at U:\Documents\eclipse\workspace_Vaadin\my-app\node_modules\enhanced-resolve\lib\Resolver.js:209:21
    at U:\Documents\eclipse\workspace_Vaadin\my-app\node_modules\enhanced-resolve\lib\Resolver.js:285:5
    at eval (eval at create (U:\Documents\eclipse\workspace_Vaadin\my-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
```
(again - full log included: log_building_app_with_production_flag.txt).

Any wisdom how I could get any Vaadin app building and running again?
This is really weird! I had no such issues whatsoever when I started using and developing with Vaadin about 1/2 year ago. And all of a sudden - poof - NOTHING is working any more. :-( 

Michael

[log_running_my_app.txt](https://github.com/vaadin/flow/files/6602684/log_running_my_app.txt)
[log_building_app_with_production_flag.txt](https://github.com/vaadin/flow/files/6602686/log_building_app_with_production_flag.txt)

**Comment by knoobie** [comment]
There is a lot going on..  Just an idea because I'm not really an expert with the frontend build.. could it be possible that you updated your vaadin version and it changes from `npm` to `pnpm` as default build? I know that the `node_modules` isn't compatible with both technologies.. this could explain why it went mayhem.  The logs are totally crazy for me.. it tries to use `npm` and later on `pnpm` or am I misinterpreting it? Can you share a screenshot / list of files and folders next to your pom to see what files are there, that could create such situation? 


At least this problem `Error: Can't resolve 'lumo-css-framework/all-classes.css'` could be fixed.. with my knowledge..looks like you have a missing frontend dependency. There is an issue about this problem https://github.com/vaadin/flow/issues/10972 

You can try the following in the meantime: 
 - add the following java class and try to `mvn clean install` again, this should add and pin the correct dependency

```java

import com.vaadin.flow.component.dependency.NpmPackage;

// Temporary until lumo-css-framework is part of Vaadin 21
@NpmPackage(value = "lumo-css-framework", version = "3.0.14")
public class VaadinNpmConfig {
}

```

**Comment by mmoser18** [comment]
As I wrote: this is a Hello-World application that I generated and downloaded yesterday from the Vaadin home page's application generator (https://start.vaadin.com/)? I didn't change a single bit in it and tried to build it as-is! And that already failed out-of-the-box.

I am trying to get this building and running to identify the issue at hand. Next step will then be - that's at least my hope - I will try to apply the fix(es) I find to get my other two actual applications up and running again.

**Later addition:**

Bingo! The suggested fix/addition got this application building and also running again without error. 
Thanks a lot! That was great support!!

So, that application generator should really be updated accordingly to spare future users similar frustrations!

**Comment by knoobie** [comment]
@mmoser18 the following property in the application.properties should force Vaadin to use npm "vaadin.pnpm.enable=false"

**Comment by mmoser18** [comment]
@knoobie I already tried that - it did not work. In production mode this file gets regenerated (with vaadin.pnpm.enable=true) and then the packaging starts so fast that there is no time to change that value. In dev-mode this flag is missing. I tried and added it before starting the application but had no effect. The frontend building still tried to install pnpm locally and then hung again.

**Comment by knoobie** [comment]
Wait.. Vaadin does not change the application.properties/yml file anyhow, are you sure we are talking about the same file? 

BTW is there a way to update your 18.0.2 to 19 or even 20? I think this old version doesn't support newer pnpm versions.. I know that there was a problem with newer node and npm/pnpm versions in the past that was fixed in newer versions of the framework.

**Comment by mmoser18** [comment]
> Vaadin does not change the application.properties
Not the properties files but the generated `flow-build-info.json` file which according to the Troubleshooting guide controls the frontend generation.

> is there a way to update your 18.0.2 to 19 or even 20
I tried with v20 yesterday and it showed exactly the same behavior, i.e. it hung during frontend compilation/building :-(

**Comment by knoobie** [comment]
The problem with pnpm > 5 is, that vaadin currently does not support it.. 

https://github.com/vaadin/flow/blob/29f317ceb61e190d4e69330cf6eeb5e0c913c5f0/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendTools.java#L123-L128

So there isn't really anything you can do to get this pnpm version working at the moment.. only way would be a downgrade to an older version like e.g. 5.x 

I'm not much help with the hanging process.. I only had this problem once in my company with a proxy in between with slowed the build to >= 20min..

**Comment by mmoser18** [comment]
Proxy or network speed are certainly not the issue here. I have a fast connection and no proxy in place.

What I don't get: I was perfectly able to build and execute Vaadin apps till last Thursday. I had node.js v14.17.0 installed on my system, although I didn't even know that Vaadin uses that, if it finds it installed (and - frankly - I hadn't even cared till then).
Something strange must have happened then, because the first time a build failed it was because of some endless recursion during frontend build (https://stackoverflow.com/questions/67828934/recursion-in-frontend-compilation-how-to-analyze-and-debug-this?noredirect=1#comment119899683_67828934).
When I couldn't resolve that I removed all generated stuff from my project, essentially leaving the src- and frontend-folders and the pom.xml and then I tried to rebuild again. And since then I can't build at all anymore. 

I spent Friday and almost the entire weekend on this and it drives me nuts! How could this have worked before and now insist on using an old pnpm version that IMHO can't have been there before???

**Comment by knoobie** [comment]
That probably happened because of an auto-update of node or pnpm on your system.. at least that is the only way I could wrap my head around it. 

I could find the following matrix, meaning node 10 would be the version that does not have pnpm and all other (>10) can at least work with it.. https://pnpm.io/installation#compatibility I don't think they switch.. it just auto-updates itself.. but the whole JS world is voodoo for me as well. I would probably try to remove pnpm locally and globally and let vaadin try to install it by itself.. hopefully that works

**Comment by mmoser18** [comment]
> let vaadin try to install it by itself.. hopefully that works
 That doesn't work either. I had tried that during the process, i.e. I had uninstalled node.js (and with it npm, pnpm, chocolatey, and what not). Then I removed ALL remainings of it that I could find on my system (AppData\npm, ...\npm-cache, ...\chocolatey stuff,  etc.). 
Then I did the vaadin dance and had Vaadin install itself what it needs. And guess what? It downloaded and installed node.js 14.17.0 (and with it pnpm 6.7.1) :-(

**Comment by knoobie** [comment]
I'm out of ideas as well.. need to wait for the vaadiners tomorrow and hopefully they have an idea

**Comment by mmoser18** [comment]
Yeah - right - it's also late. Thanks for helping that far!

**Comment by pleku** [comment]
There is so much going on that it is difficult to even start.

> I'll try to locate an node.js version that includes pnpm v<6 and install that. You don't happen to know when node.js switched to a pnpm v>6.0, do you?

Node.js comes with npm out-of-the-box. But never with pnpm - this is installed by Vaadin.
When you use pnpm, you should never update it manually. In case an unsupported pnpm version is detected, then the build will try to install a version that is supported.

Vaadin version 18 is not even supported anymore, so I'd recommend that you update your project to latest `20.0.1` and we focus on issues with that. Then you either use npm or pnpm, it doesn't matter, but you don't need to delete anything as the flow build should be able do to that when necessary.

Then I would appreciate if you can give out the list of the versions you are using for the project (running with Vaadin 20.0.1 or newer): Java, node + npm/pnpm (depending on which you're trying to run with) and the latest error logs for the build.
Even better would be the project to reproduce the issue with, you can try a fresh project from https://start.vaadin.com/?preset=latest and see if that works.

**Comment by Haprog** [comment]
One note about the "Vaadin dance". You should never remove `package.json` (at least from any recent project generated from start.vaadin.com or in any project where you have manually installed any dependencies via npm or pnpm). Otherwise you will lose some needed client side dependencies and things will not work.

> node.js v14.16.0 (from Feb. 2021) also already contains pnpm v6.7.1. I just installed that one and checked.

This is just not true. Node.js has never included any version of pnpm by default. They only include npm. I'm not sure if the globally installed packages persist in case you uninstall/reinstall Node.js.

pnpm is normally installed via npm and you can install any specific version of it to upgrade/downgrade if you want to try it out. Assuming you have Node.js installed globally, and you want to install a specific version of pnpm globally then you can run:
```
npm i -g pnpm@5
```
(or replace the `5` with any version, this just gets the latest v5 but you can specify exact version like `@1.2.3`)

To remove pnpm (if it's installed globally):
```
npm uninstall pnpm
```

I'm not sure but I think Vaadin doesn't actually "install" pnpm if it's not already installed as it just runs it with `npx` which means if it's not installed, it will download and run it without installing (the downloaded version may be cached though). If you do have `pnpm` explicitly installed, then I think it will just use the installed version (at least if it's compatible). At least in a V20 project Vaadin tries to run pnpm version 5 explicitly (`pnpm@5` via npx). See also this for some more info: https://vaadin.com/docs/v20/guide/install/#pnpm

I just tested getting a new V20 Flow project from https://start.vaadin.com/ on a Win 10 machine that doesn't have Node.js (or npm and pnpm) globally installed and just running `mvn` in the project worked fine. Vaadin installed Node.js v14.15.4 into `.vaadin` and I can see it installed the frontend dependencies with pnpm 5:
```
2021-06-07 11:22:08.757  INFO 21400 --- [onPool-worker-3] c.v.flow.server.frontend.FrontendTools   :
  using 'C:\Users\hapro\.vaadin\node\node.exe C:\Users\hapro\.vaadin\node\node_modules\npm\bin\npx-cli.js --yes --quiet pnpm@5' for frontend package installation
2021-06-07 11:24:11.228  INFO 21400 --- [onPool-worker-3] c.v.f.s.frontend.TaskUpdatePackages      : 
  Frontend dependencies resolved successfully.
```

**Comment by mmoser18** [comment]
* you should never remove package.json 

OK - thanks for pointing that out. Will remove it from my extended `vaadin_dance.cmd` then.

* This is just not true. Node.js has never included any version of pnpm by default.

Well - then I don't know, *how* this got installed, but each time after I had installed node.js (I had tried with v14.17.0 and v14.16.0) there was a `pnpm.cmd` and `pnmp.ps1` underneath `C:\Program Files\node` which - when I issued a `pnpm --version` identified itself as v6.7.1. And I can assure you that I had double checked that this had been deleted before! I can send you the mentioned `vaadin_dance.cmd` that I had always run between my attempts. It definitely takes care that `C:\Program Files\nodejs` and everything underneath is removed. And I had also double checked that this file exists nowhere on my entire disk (I have a tool for that).

* To remove pnpm...  

I had tried that path as well and had manually uninstalled pnpm (I actually had to issue `npm -global uninstall pnmp`) and then installed pnpm@4.4.0 (because that's what the error message suggested). While it did install fine (and there was then also no message that the build was going to install v4.4.0 locally, so I assume it *did* pick it up) the builds nevertheless all hung as well.

* I just tested getting a new V20 (Java 8, Flow) ... Vaadin installed Node.js v14.15.4

That's really odd, since in my attempts over the weekend it always installed v14.17.0. Are you sure there is no cache or proxy in between?

**Comment by pleku** [comment]
The code shipped in Vaadin 20 at least still points to v14.5.4 as the default version to be installed https://github.com/vaadin/flow/blob/7.0/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendTools.java#L55, even though v14.17.0 is the latest LTS.

I just installed v14.17.0 on my local Windows system and got a fresh project from start.vaadin.com running Vaadin 20.0.1 in hybrid mode (with both Flow + Fusion views). Running `mvn spring-boot:run` everything works as expected.

After I switched to npm, and ran `mvn clean spring-boot:run`, I ran into a new issue https://github.com/vaadin/flow/issues/11177#issuecomment-856702221 which is happening only on Windows when you swap from pnpm -> npm. After running the project again, it starts to work.

So to be able to proceed with this issue, we would need a way to reproduce it. At the moment I can't give any further hints with this.

**Comment by Haprog** [comment]
> Well - then I don't know, how this got installed, but each time after I had installed node.js (I had tried with v14.17.0 and v14.16.0) there was a pnpm.cmd and pnmp.ps1 underneath C:\Program Files\nodejs\... which - when I issued a pnpm --version identified itself as v6.7.1.

@mmoser18 That is very weird. Did you install Node.js via the official installer from https://nodejs.org/en/download/ or some other means? I just tried installing https://nodejs.org/dist/v14.17.0/node-v14.17.0-x64.msi and this is what I have in `C:\Program Files\nodejs`
![image](https://user-images.githubusercontent.com/75155/121190429-b9dd1480-c873-11eb-955b-6af52eb147e3.png)
No `pnpm` only `npm`.

**Comment by mmoser18** [comment]
After another de-install and removal of everything node/npm/pnpm I started over from a freshly created and downloaded HelloWorld application (using Vaadin 20 this time) and that finally worked. I then started re-adding all src and frontend stuff from my old project and luckily it kept building and working. So meanwhile I am back in business.
Still beats me what exactly caused this and - since I couldn't really pinpoint the issue - I am afraid it might happen again, but left's keep fingers crossed.

**Comment by mmoser18** [comment]
Just in case anyone find this and has the same problem: I most likely could have spared me the entire dance and the hassle. When this same situation happened a second time I asked on StackOverflow where I got an answer (https://stackoverflow.com/questions/68216553/vaadin-again-build-completely-stalled) that explained the issue and also pointed me the way how to fix it.
